### PR TITLE
Calendar: added new week view for calendar

### PR DESCRIPTION
**Pull request by OwlsForever**
image

### CODE REVIEW

**Comment by OwlsForever** [comment]
[Display CW calendar - weekly view](https://trello.com/c/oWRv8uid/2178-display-cw-calendar-weekly-view)

**Comment by artxp** [comment]
https://github.com/area9innovation/nnc/pull/1321

**Review by artxp — APPROVED** [review]
LGTM, although duplicates code

**Comment by sergeymaksimov** [comment]
@OwlsForever As I can see there are not many changes. Maybe it is possible to add some styles to CalendarForEventsStyle that will change current week view to minimize duplicate code?

**Comment by OwlsForever** [comment]
@sergeymaksimov done

File: lib/material/extra/calendar/calendar.flow [hunk 1]
```diff
@@ -3664,7 +3741,11 @@ CalendarForEvents(
 				}));
 
 				headerTextM = MSelect(currentDate, \date -> {
-					defaultText = formatString("WEEK %1 OF %2", [i2s(getWeekOfYear(date, 0, true)), i2s(date.year)]);
+					defaultText = if (containsStruct(styles, ShowHoursGridForWeekView(0.0))) {
```
Comment (sergeymaksimov) [inline lib/material/extra/calendar/calendar.flow:3744]: Could you use CalendarHeaderText style in nnc to change text?

Comment (OwlsForever) [reply lib/material/extra/calendar/calendar.flow]: Done. Anything else?

**Review by sergeymaksimov — APPROVED** [review]

**Pull request merged**